Source/HarmonyPatches/Patches_PathFinding/Patch_InjectLockCheck.cs

```diff
@@ -0,0 +1,57 @@
+﻿using RimWorld;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+using System.Text;
+using HarmonyLib;
+using Verse;
+using Verse.AI.Group;
+using PrisonLabor.Core.LaborArea;
+
+namespace PrisonLabor.HarmonyPatches.PathFinding
```

Comment:
Hope it will not mess with mod Locks and still works without mod :P

Reply:
Easy to add a compatibility patch.

Reply:
Added in https://github.com/kbatbouta/PrisonLabor/commit/ac9636cbe939069cc7a618c805eea67cb9d6ae16